tesseract_common/test/tesseract_common_unit.cpp

```diff
@@ -236,6 +242,277 @@ TEST(TesseractCommonUnit, bytesResource)
+
+  {
+    std::ofstream os("/tmp/tool_center_point_boost.xml");
+    boost::archive::xml_oarchive oa(os);
+    oa << BOOST_SERIALIZATION_NVP(tcp);
+  }
+
+  tesseract_common::ToolCenterPoint ntcp;
+  {
+    std::ifstream ifs("/tmp/tool_center_point_boost.xml");
+    assert(ifs.good());
+    boost::archive::xml_iarchive ia(ifs);
+
+    // restore the schedule from the archive
+    ia >> BOOST_SERIALIZATION_NVP(ntcp);
+  }
+
+  EXPECT_TRUE(tcp == ntcp);
+}
+
+TEST(TesseractCommonUnit, serializationManipulatorInfo)
+{
+  Eigen::Isometry3d pose = Eigen::Isometry3d::Identity();
+
+  tesseract_common::ManipulatorInfo manip_info("manipulator");
+  manip_info.tcp = tesseract_common::ToolCenterPoint(pose);
+
+  {
+    std::ofstream os("/tmp/manipulator_info_boost.xml");
+    boost::archive::xml_oarchive oa(os);
+    oa << BOOST_SERIALIZATION_NVP(manip_info);
+  }
+
+  tesseract_common::ManipulatorInfo nmanip_info;
+  {
+    std::ifstream ifs("/tmp/manipulator_info_boost.xml");
+    assert(ifs.good());
+    boost::archive::xml_iarchive ia(ifs);
+
+    // restore the schedule from the archive
+    ia >> BOOST_SERIALIZATION_NVP(nmanip_info);
+  }
+
+  EXPECT_TRUE(manip_info == nmanip_info);
+}
+
+TEST(TesseractCommonUnit, serializationJointState)
+{
+  tesseract_common::JointState joint_state;
+  joint_state.joint_names = { "joint_1", "joint_2", "joint_3" };
+  joint_state.position = Eigen::VectorXd::Constant(3, 5);
+  joint_state.velocity = Eigen::VectorXd::Constant(3, 6);
+  joint_state.acceleration = Eigen::VectorXd::Constant(3, 7);
+  joint_state.effort = Eigen::VectorXd::Constant(3, 8);
+  joint_state.time = 100;
+
+  {
+    std::ofstream os("/tmp/joint_state_boost.xml");
+    boost::archive::xml_oarchive oa(os);
+    oa << BOOST_SERIALIZATION_NVP(joint_state);
+  }
+
+  tesseract_common::JointState njoint_state;
+  {
+    std::ifstream ifs("/tmp/joint_state_boost.xml");
+    assert(ifs.good());
+    boost::archive::xml_iarchive ia(ifs);
+
+    // restore the schedule from the archive
+    ia >> BOOST_SERIALIZATION_NVP(njoint_state);
+  }
+
+  EXPECT_TRUE(joint_state == njoint_state);
+}
+
+TEST(TesseractCommonUnit, serializationKinematicLimits)
+{
+  tesseract_common::KinematicLimits limits;
+  limits.joint_limits.resize(3, 2);
+  limits.joint_limits << -5, 5, -5, 5, -5, 5;
+  limits.velocity_limits = Eigen::VectorXd::Constant(3, 6);
+  limits.acceleration_limits = Eigen::VectorXd::Constant(3, 7);
+
+  {
+    std::ofstream os("/tmp/kinematic_limits_boost.xml");
+    boost::archive::xml_oarchive oa(os);
+    oa << BOOST_SERIALIZATION_NVP(limits);
+  }
+
+  tesseract_common::KinematicLimits nlimits;
+  {
+    std::ifstream ifs("/tmp/kinematic_limits_boost.xml");
+    assert(ifs.good());
+    boost::archive::xml_iarchive ia(ifs);
+
+    // restore the schedule from the archive
+    ia >> BOOST_SERIALIZATION_NVP(nlimits);
+  }
+
+  EXPECT_TRUE(limits == nlimits);
+}
+
+TEST(TesseractCommonUnit, serializationVectorXd)
```

Comment:
A couple of additional tests that might be nice. I think these would pass, but they would catch behavior changes in the future.
- Serializing an empty object
- Deserializing to an already filled object
- Deserializing to an object that is the incorrect size

Reply:
Done.

Reply:
I did the same for MatrixX2d